From ca94f4474ebbe062af65425436dcaf71e50427ec Mon Sep 17 00:00:00 2001
From: Shinya Kuribayashi <shinya.kuribayashi@necel.com>
Date: Mon, 7 Sep 2009 09:21:16 +0900
Subject: [PATCH 13/18] MIPS: nec_emma: Do not assume prom_{argc,argv,envp} are non-zero/non-NULL

These variables easily could be zero/NULL, for example, when you work with
a JTAG-ICE debugger.

Signed-off-by: Shinya Kuribayashi <shinya.kuribayashi@necel.com>
---
 arch/mips/emma/common/prom.c |    7 +++++++
 1 files changed, 7 insertions(+), 0 deletions(-)

diff --git a/arch/mips/emma/common/prom.c b/arch/mips/emma/common/prom.c
index 12eb89b..b16df0a 100644
--- a/arch/mips/emma/common/prom.c
+++ b/arch/mips/emma/common/prom.c
@@ -48,6 +48,10 @@ void __init prom_init_cmdline(void)
 	char *cp;
 	int actr;
 
+	/* Make sure arguments are available.  If not, do nothing. */
+	if (prom_argc == 0 || prom_argv == NULL)
+		return;
+
 	actr = 1; /* Always ignore argv[0] */
 
 	cp = &(arcs_cmdline[0]);
@@ -68,6 +72,9 @@ char * __init prom_getenv(char *envname)
 {
 	int index, len, yamon;
 
+	if (!prom_envp)
+		return NULL;
+
 	index = 0;
 	len = strlen(envname);
 	yamon = (prom_envp[0] && strchr(prom_envp[0], '=') == NULL);
-- 
1.6.3.3

